### PR TITLE
Patient operator

### PR DESCRIPTION
**Pull request by elliotdes**
Adds a `FivetranPatientOperator` that waits for the Fievtran sync to finish before marking the task as completed. This is necessary for certain instances in which parallelism is limited.

### CODE REVIEW

**Comment by PubChimps** [comment]
thank you @elliotdes, looking in this https://github.com/fivetran/airflow-provider-fivetran/pull/56 now

**Review by denimalpaca — CHANGES_REQUESTED** [review]
Few minor changes to the initializer would be nice. Not sure this is really standard practice for operators to wait like this, it's really more of sensor behavior I think. If you think there's a really good use case for this, then by all means go for it. Also happy to run this by the Astronomer team at large and get a sense of whether there's another way that might be better suited to operators.

File: fivetran_provider/operators/fivetran.py [hunk 1]
```diff
@@ -81,3 +83,93 @@ def execute(self, context):
...
+    operator_extra_links = (RegistryLink(),)
+
+    # Define which fields get jinjaified
+    template_fields = ["connector_id"]
+
+    @apply_defaults
+    def __init__(
+        self,
+        connector_id: str,
+        timeout_seconds: int,
```
[the remark this answers is not in the document]
Comment (elliotdes) [reply fivetran_provider/operators/fivetran.py]: Done!

File: fivetran_provider/operators/fivetran.py [hunk 2]
```diff
@@ -81,3 +83,93 @@ def execute(self, context):
...
+        run_name: Optional[str] = None,
+        fivetran_conn_id: str = "fivetran",
+        fivetran_retry_limit: int = 3,
+        fivetran_retry_delay: int = 1,
+        poll_frequency: int = 15,
+        manual: bool = True,
```
[the remark this answers is not in the document]
Comment (elliotdes) [reply fivetran_provider/operators/fivetran.py]: Done, now matches the order used in the `FivetranOperator`

**Comment by elliotdes** [comment]
@denimalpaca we have to use this operator in our DAG's due to limited parallelism in our Airflow instance.

The `FivetranSensor` works by checking the fivetran connectors initial status against future pokes. If the `FivetranSensor` begins after the connector has already finished its sync, then the sensor will hang. This is what was happening for us.

This `FivetranPatientOperator` would not be needed if the Fivetran API had unique id's for each triggered run that could then be checked explicitly. This is how [dbt Cloud](https://docs.getdbt.com/dbt-cloud/api-v2#operation/getRunById), and that works quite well for us.

**Comment by kieronellis** [comment]
+1 on this - we are also experiencing this issue where the FivetranSensor starts after the connector has already finished its sync and so the FivetranSensor keeps reporting `sync_state = scheduled`.

**Comment by denimalpaca** [comment]
@PubChimps do you have any input here on using the Sensor to avoid starting after the connector has finished? My only thought is to pattern the DAG differently: have the sensor and connector run in parallel, then set the next task to `trigger_rule=all_success`.

**Comment by elliotdes** [comment]
@denimalpaca is there a way to guarantee two tasks running in parallel? If we take an extreme example where we have parallelism set to 1 on Airflow, then im not sure if the both tasks would trigger at the same time

**Comment by elliotdes** [comment]
@denimalpaca would this operator be worth committing for people limited to 1 parallelism / Sequential Operator then?

**Comment by denimalpaca** [comment]
@elliotdes ideally people aren't running production pipelines with a sequential executor, but for those who are, yes this will probably be useful!

**Review by denimalpaca — APPROVED** [review]

**Comment by PubChimps** [comment]
Hi all, sorry for the delays here, but could you please check v 1.1.1 of the provider? @kieronellis this will solve the issues that you mentioned you have experienced

**Pull request closed without merging**